### PR TITLE
#166265845 users can signup

### PR DESCRIPTION
**Pull request by Hadeneekeh**
#### What does this PR do?
It has the User can signup and receives JWT

#### Description of Task to be completed?
- Write unit test
- create the signup endpoint
- integrate JWT

#### How should this be manually tested?
* clone the repo git clone `https://github.com/emp-daisy/HelloBooksLib.git`
* on your terminal run `cd HelloBooksLib`
* run `npm install`
* visit the [staging app link](https://helobooks-staging.herokuapp.com/api/v1) @https://helobooks-staging.herokuapp.com/api/v1 and test with postman

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
[#166265845](https://www.pivotaltracker.com/story/show/166265845)

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const UserController = {
+    async signUp(req, res) {
+        models.User.findOne({
+            where: { email: req.body.email }
+          })
+          .then((foundUser) => {
+            if (foundUser) {
+              return res.status(409).send({
+               status: res.statusCode,
+               message: 'A user with that email address exist already',
```
Comment (chuxmykel) [inline server/controllers/userController.js]: "email address `exists` already"

File: server/db/migrations/20190528151100-create-user.js [hunk 2]
```diff
@@ -1,37 +1,38 @@
-/* eslint-disable no-unused-vars */
```
Comment (chuxmykel) [inline server/db/migrations/20190528151100-create-user.js]: Did you rebase? You are undoing a lot.

File: server/db/seeders/20190528204615-demo-user.js [hunk 3]
```diff
@@ -1,13 +1,16 @@
-/* eslint-disable no-unused-vars */
```
Comment (chuxmykel) [inline server/db/seeders/20190528204615-demo-user.js]: You need to check these files. You are overwriting the ES6 files with the old ES5 version

File: server/routes/index.js [hunk 4]
```diff
@@ -1,15 +1,12 @@
 import express from 'express';
-import swaggerUI from 'swagger-ui-express';
```
Comment (chuxmykel) [inline server/routes/index.js]: Why are you removing the docs?

**Review by chuxmykel — DISMISSED** [review]

File: server/db/migrations/20190528151100-create-user.js [hunk 5]
```diff
@@ -1,37 +1,38 @@
...
-  });
-}
-export function down(queryInterface, Sequelize) {
-  return queryInterface.dropTable('Users');
-}
\ No newline at end of file
+'use strict';
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Users', {
```
Comment (encodedBicoding) [inline server/db/migrations/20190528151100-create-user.js]: How did we get back here? I think u should do a rebase

File: server/controllers/userController.js (hunk 1 above)
Comment (Hadeneekeh) [reply server/controllers/userController.js]: Okay.

File: server/db/migrations/20190528151100-create-user.js (hunk 2 above)
Comment (Hadeneekeh) [reply server/db/migrations/20190528151100-create-user.js]: Yes, I did.

**Review by encodedBicoding — APPROVED** [review]

File: server/db/migrations/20190528151100-create-user.js (hunk 5 above)
Comment (Hadeneekeh) [reply server/db/migrations/20190528151100-create-user.js]: That file is marked outdated, I have made corrections.

**Review by OvieMudi — APPROVED** [review]

**Review by chuxmykel — APPROVED** [review]
LGTM

File: server/db/migrations/20190528151100-create-user.js (hunk 5 above)
Comment (chuxmykel) [reply server/db/migrations/20190528151100-create-user.js]: She has fixed this @encodedBicoding

**Pull request merged**